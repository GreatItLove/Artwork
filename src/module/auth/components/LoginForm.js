import React from 'react';
import { Field } from 'redux-form';
import { Link } from 'react-router-dom';
import MenuItem from 'material-ui/MenuItem';
import PropTypes from 'prop-types';

import SelectField from '../../common/components/form/SelectField';
import InputField from '../../common/components/form/InputField';
import SubmitButton from '../containers/SubmitButton';
import { FORGOT_PASSWORD_PATH } from '../../../config/routesPatch';
import * as routes from '../../../config/routesPatch';

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  login: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired
};
/**
 * @function LoginForm
 * @param {Function} handleSubmit
 * @param {Function} login
 * @param {Function} t translate function
 * @returns {XML}
 */
function LoginForm({ handleSubmit, login, t }) {
  return (
    <div>
      <div className="auth-login-wrapper">
        <div className="auth-login-fields">
          <div className="py-2 d-md-none" />
          <h4>{t('login.formHeader')}</h4>
          <div className="py-2 d-md-none" />
          <form onSubmit={handleSubmit(login)}>
            <Field
              id="name"
              name="username"
              floatingLabelText={t('login.emailLabel')}
              fullWidth={true}
              component={InputField}
              autoComplete="new-username"
            />
            <Field
              id="pass"
              name="password"
              component={InputField}
              floatingLabelText={t('login.passwordLabel')}
              type="password"
              autoComplete="new-password"
            />
            <Field
              id="start-page"
              name="startPage"
              floatingLabelText={t('login.startPageField')}
              fullWidth={true}
              component={SelectField}
            >
              <MenuItem value={routes.ADMIN_PANEL_BASE_PATH} primaryText={t('login.startPageOption.home')} />
              <MenuItem value={routes.ARTISTS_PATH} primaryText={t('login.startPageOption.artists')} />
              <MenuItem value={routes.CONSIGNMENTS_PATH} primaryText={t('login.startPageOption.consignments')} />
              <MenuItem value={routes.ARTWORK_PATH} primaryText={t('login.startPageOption.artwork')} />
              <MenuItem value={routes.ARTWORK_PATH} primaryText={t('login.startPageOption.artworkAddWizard')} />
              <MenuItem value={routes.CONTACTS_PATH} primaryText={t('login.startPageOption.contacts')} />
              <MenuItem value={routes.CONTACTS_PATH} primaryText={t('login.startPageOption.contactsAddWizard')} />
              <MenuItem value={routes.WEBSITES_PATH} primaryText={t('login.startPageOption.websiteSection')} />
              <MenuItem value={routes.EXIBITIONS_PATH} primaryText={t('login.startPageOption.exhibitions')} />
              <MenuItem value={routes.RESOURCES_PATH} primaryText={t('login.startPageOption.resources')} />
              <MenuItem value={routes.INVOICES_PATH} primaryText={t('login.startPageOption.invoices')} />
              <MenuItem value={routes.QUICK_SALE_PATH} primaryText={t('login.startPageOption.quickSale')} />
              <MenuItem value={routes.VISITOR_TODAY_PATH} primaryText={t('login.startPageOption.visitorToday')} />
              <MenuItem value={routes.SURVEY_RESULTS_PATH} primaryText={t('login.startPageOption.surveyResults')} />
            </Field>
            <div className="pt20">
              <SubmitButton label={t('login.submitBtn')} />
            </div>
            <div className="pt20 links">
              <Link to={FORGOT_PASSWORD_PATH}>{t('login.forgotPassword')}</Link>
              <span>|</span>
              <a href="http://www.managedartwork.com/Demo.cfm" target="_blanko">
                {t('login.register')}
              </a>
            </div>
          </form>
          <div className="py-2 d-md-none" />
        </div>
      </div>
    </div>
  );
}

LoginForm.propTypes = propTypes;

export default LoginForm;
