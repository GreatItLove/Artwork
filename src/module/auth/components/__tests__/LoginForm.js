/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import LoginForm from '../../components/LoginForm';

describe('Login component', () => {
  const handleSubmit = jest.fn();
  const login = jest.fn();
  const t = key => key;
  const wrapper = shallow(<LoginForm {...{ handleSubmit, login, t }} />);
  it('Shallow snapshot initial state ', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
