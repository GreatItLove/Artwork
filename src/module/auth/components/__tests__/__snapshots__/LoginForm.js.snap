// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Shallow snapshot initial state  1`] = `
<div>
  <div
    className="auth-login-wrapper"
  >
    <div
      className="auth-login-fields"
    >
      <div
        className="py-2 d-md-none"
      />
      <h4>
        login.formHeader
      </h4>
      <div
        className="py-2 d-md-none"
      />
      <form>
        <Field
          autoComplete="new-username"
          component={[Function]}
          floatingLabelText="login.emailLabel"
          fullWidth={true}
          id="name"
          name="username"
        />
        <Field
          autoComplete="new-password"
          component={[Function]}
          floatingLabelText="login.passwordLabel"
          id="pass"
          name="password"
          type="password"
        />
        <Field
          component={[Function]}
          floatingLabelText="login.startPageField"
          fullWidth={true}
          id="start-page"
          name="startPage"
        >
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.home"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.artists"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/artists"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.consignments"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/consignments"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.artwork"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/artwork"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.artworkAddWizard"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/artwork"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.contacts"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/contacts"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.contactsAddWizard"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/contacts"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.websiteSection"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/website"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.exhibitions"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/exhibitions"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.resources"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/resources"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.invoices"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/invoices"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.quickSale"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/quick-sale"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.visitorToday"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/visitor-today"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="login.startPageOption.surveyResults"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="/survey-results"
          />
        </Field>
        <div
          className="pt20"
        >
          <Connect(SubmitBtnLoader)
            label="login.submitBtn"
          />
        </div>
        <div
          className="pt20 links"
        >
          <Link
            replace={false}
            to="/forgot-password"
          >
            login.forgotPassword
          </Link>
          <span>
            |
          </span>
          <a
            href="http://www.managedartwork.com/Demo.cfm"
            target="_blanko"
          >
            login.register
          </a>
        </div>
      </form>
      <div
        className="py-2 d-md-none"
      />
    </div>
  </div>
</div>
`;
